src/Image.js

```diff
@@ -68,16 +68,16 @@ function ImageBase({
   imgProps = {}, 
   layoutClassName = undefined 
 }) {
-  if (process.env.NODE_ENV !== 'production' && !image.asset.metadata) {
-    console.error('Image asset doesn\'t have associated metadata. Did you forget to dereference the asset field (`image{..., asset->}`)?')
+  if (image?.asset && !image.asset.metadata) {
+    throw new Error('Image asset doesn\'t have associated metadata. Did you forget to dereference the asset field (`image{..., asset->}`)?')
   }
 
   const { ref: sizeRef, size: displaySize } = useElementSize()
   const { src, srcSet } = useSrcSet({ config: sanityConfig, image, adjustImage })
   const { width, height, size } = useDerivedSizes({
     deriveSizes, 
     displaySize,
-    naturalSize: image.asset.metadata.dimensions  
+    naturalSize: image.asset?.metadata.dimensions ?? { width: 0, height: 0 }
```

Comment:
Als `asset.metadata` er is... is `asset.metadata.dimensions` er dan ook gegarandeerd?

```suggestion
    naturalSize: image.asset?.metadata?.dimensions ?? { width: 0, height: 0 }
```

Reply:
Yes :)

> There are three additional metadata options that are always included and cannot be disabled: `dimensions`, `hasAlpha`, and `isOpaque`. 

Maar door je comment zie ik wel nog een andere: omdat we een error throwen als `image.asset` of `image.asset.metadata` er niet is, kan dit `image.asset.metadata.dimensions` worden

Reply:
@peeke 

> omdat we een error throwen als image.asset of `image.asset.metadata` er niet is, kan dit `image.asset.metadata.dimensions` worden

Helaas is dat niet zo. Als `image.asset` leeg is gaat dit een error geven. Je gooit geen error als `image.asset` leeg is. Hoogstwaarschijnlijk om te zorgen dat `?preview` goed gaat.

Reply:
Right, jullie hebben helemaal gelijk :+1: